Sources/MapboxMaps/Foundation/MapEventHandler.swift

```diff
@@ -0,0 +1,64 @@
+import Foundation
+
+public protocol MapEventsObservable: AnyObject {
+    /// Listen to Map events.
+    ///
+    /// - Parameters:
+    ///   - eventType: The event type to listen to.
+    ///   - handler: The block of code to execute when the event occurs. Return
+    ///     `true` to indicate that you have handled the event(s) and no longer
+    ///     wish to receive them.
+    ///
+    /// - Returns: A `Cancelable` object that you can use to stop listening for
+    ///     events, in the case your closure does not return `true`.
+    @discardableResult
+    func on(_ eventType: MapEvents.EventKind, handler: @escaping (Event) -> Bool) -> Cancelable
```

Comment:
Is using a `bool` here as a return expressive enough? It feels a little obscure.. Maybe we should have two methods like so:
```swift
// This is a one-shot subscription
func onNext(_ eventType: MapEvents.EventKind, handler: @escaping (Event) -> Void)

// This is a many-shot subscription
func onEvery(_ eventType: MapEvents.EventKind, handler: @escaping (Event) -> Void) -> Cancelable
```